spec/src/modules/tracker.js

```diff
@@ -3915,19 +4146,69 @@ describe('ConstructorIO - Tracker', () => {
     const requiredParameters = {
       items: [
         {
-          item_id: 'labradoodle',
-          variation_id: 'labradoodle-black',
+          itemId: 'labradoodle',
+          variationId: 'labradoodle-black',
         },
         {
-          item_id: 'product55f1b3577fa84947a93ea01b91d52f45',
+          itemId: 'product55f1b3577fa84947a93ea01b91d52f45',
         },
       ],
       revenue: 123.45,
     };
     const optionalParameters = {
-      order_id: '123938123',
+      orderId: '123938123',
       section: 'Products',
     };
+    const snakeCaseItems = [
+      {
+        item_id: "labradoodle",
+        variation_id: "labradoodle-black",
+      },
+      {
+        item_id: "product55f1b3577fa84947a93ea01b91d52f45",
+      },
+    ];
+
+    it('Backwards Compatibility - Should respond with a valid response when snake cased parameters are provided', (done) => {
+      const { tracker } = new ConstructorIO({
+        apiKey: testApiKey,
+        fetch: fetchSpy,
+      });
+      const snakeCaseParameters = {
+        order_id: '123938123',
+        revenue: 123.45,
+        items: snakeCaseItems,
+      }
+
+      tracker.on('success', (responseParams) => {
+        const requestParams = helpers.extractBodyParamsFromFetch(fetchSpy);
+        
+        try {
+        // Request
+        expect(fetchSpy).to.have.been.called;
```

Comment:
Is this code in this try block missing indentations?

Also, what prompted adding the try/catch blocks? 🤔 Do we need them?

Reply:
We need to call done() when tests are done but if any assertions fail then done() never gets called, the test fails when the timeout is reached.

When I was debugging, I added this so that it would catch the error, pass it to done, print it in the console, otherwise it would just timeout, fail without a message. I think we should do this for other tests too.